### ISSUE TITLE
#loaded? Appears not to work on pages without a URL defined

### ISSUE THREAD

**Issue by luke-hill** [natritmeyer/site_prism#228]
## Environment
- Operating System: Windows
- Latest versions of site_prism and capybara
- How you're interacting with the Framework: Cucumber Tests
- Browser/s Driver/s being used - with version numbers Selenium 3.4 - Any browser

## Expected Behavior
`#loaded?` would perform all load validations

## Actual Behavior
It crashes with `SitePrism::NoUrlMatcherForPage`

## Steps to reproduce
```
require 'site_prism'
require 'capybara'
require 'capybara/dsl'

class SomePage < SitePrism::Page
  element :foo_element, '.foo'
  load_validation { [has_foo_element?, 'did not have foo element!'] }
end

page = SomePage.new
page.loaded?
```

**Comment by tgaff** [comment]
Is this really a bug?  The README seems to document this as a feature.  It seems to indicate that calling `displayed?` is expected behavior.  As far as I recall displayed was originally just a URL checker.  `NOTE: SitePrism::Page includes a default load validation on page.displayed? which is applied to all pages. It is therefore not necessary to define a load validation for this condition on inheriting page objects.`

**Comment by luke-hill** [comment]
Not sure. This might be a documentation issue then. I had a re-read this morning. And on some examples a url is not defined, but then it shows as having performed this perfunctory test.

Maybe we need a new label for "Needs decision"

Ok then how about a different question.... Is this behaviour desired / desirable / the best use case?

ping @TheMetalCode for thoughts also?

**Comment by tgaff** [comment]
Perhaps the `displayed?` default load validation shouldn't be included as a default validation.  We can pass arguments to `displayed?` but since that load validation is pre-defined the user has no opportunity to pass those args.  It's always `[displayed?, "Expected #{current_url} to match #{url_matcher} but it did not."]`  I'm not sure how many people use that feature (I use regex matchers), but this might make it unusable in some cases.  

It also looks like there's no trivial way to remove that load_validation so maybe not having it would be a better default.

**Comment by tmertens** [comment]
@tgaff correct, this was included as a default validation because there are not many cases in my mind where you wouldn't have a url matcher for a page unless you simply didn't want to define one, I think it's reasonable to remove it as it is simple enough to create a base page in any project that adds this validation rather than inheriting directly from the site prism Page class.

**Comment by luke-hill** [comment]
So are we ok in thinking we "should" remove this default validation then? The case I've raised seems a good instance where you'd want a load validation on a page but wouldn't want to specify a URL (We actually have a load of these and we have an initialize method instead that is called.

**Comment by tmertens** [comment]
I am in support of it. Not sure if it requires a major version bump or not since it is a breaking change. At least initially perhaps the default validation could be disabled through configuration? Then it could be deprecated permanently at the next major version change if desired

**Comment by tgaff** [comment]
I'm ok with removing it.  However as tmertens suggests this is a breaking change.  Perhaps including it automatically only when a url matcher is defined would work?  That should avoid making this a breaking change but also allow this to work with pages without a URL matcher.  

We could consider removing it be default in the next major version.

**Comment by luke-hill** [comment]
Great points from both of you. In agreement with both

**Comment by ineverov** [comment]
@luke-hill should we deprecate it in pre-release version then? so people will have time to fix it (create a base page and inherit other pages from it)

**Comment by luke-hill** [comment]
Possibly. But this would be in a 3.0 version (Of which there will be at least 3 minor bumps before then, as there's a reasonable amount of things to still change / fix up)

**Comment by luke-hill** [comment]
I've changed this to a Feature Request (Should have done this a while ago), as we need to find a way to make it configurable first (In 2.x series), and then completely remove it in v3.0
